packages/teleterm/src/ui/TopBar/Clusters/useClusters.ts

```diff
@@ -13,6 +14,41 @@ export function useClusters() {
       .filter(c => c.leaf && c.uri.startsWith(clusterUri));
   }
 
+  function hasPendingAccessRequest() {
+    const rootClusterUri = workspacesService.getRootClusterUri();
+    const accessRequestsService =
+      workspacesService.getWorkspaceAccessRequestsService(rootClusterUri);
+    if (!accessRequestsService) {
+      return false;
+    }
+
+    const pendingAccessRequest =
+      accessRequestsService.getPendingAccessRequest();
+
+    if (!pendingAccessRequest) {
+      return false;
+    }
+
+    const count =
+      Object.keys(pendingAccessRequest.node).length +
+      Object.keys(pendingAccessRequest.db).length +
+      Object.keys(pendingAccessRequest.app).length +
+      Object.keys(pendingAccessRequest.kube_cluster).length +
+      Object.keys(pendingAccessRequest.windows_desktop).length;
```

Comment:
The same code we have in `useAccessRequestsButton`. I'd see it in the `accessRequestsService`.